File: 프로그래머스/0/120821. 배열 뒤집기/배열 뒤집기.java
class Solution {
    public int[] solution(int[] num_list) {
        int size = num_list.length;
        int[] answer = new int[size];
        
        for(int i = 0; i < size; i++) {
            answer[i] = num_list[size - 1 - i];
        }
        
        return answer;
    }
}

// 다른 풀이 1
List<Integer> list = Arrays.stream(numList).boxed().collect(Collectors.toList());
Collections.reverse(list);
return list.stream().mapToInt(Integer::intValue).toArray();

// Arrays.stream(): 배열을 스트림으로 변환
// boxed(): 기본형 스트림을 객체 스트림으로 변환 (IntStream -> Stream<Integer>)
// collect(Collectors.toList()): 스트림의 요소를 리스트에 저장하여 반환

// 다른 풀이 2
return LongStream.range(1, num_list.length + 1)
                .mapToInt(i -> num_list[(int) (num_list.length - i)])
                .toArray();
// 1부터 num_list.length + 1까지 범위의 LongStream 생성
// mapToInt를 이용해 num_list의 값 새로 세팅
// .toArray()를 이용해 배열로 반환

File: 프로그래머스/0/120821. 배열 뒤집기/배열 뒤집기.py
def solution(num_list):
    answer = []
    
    return list(reversed(num_list))

# [리스트].reverse()
# - 리스트를 뒤집는다. 
# - 값을 반환하지 않는다. 
# - 리스트 함수로 리스트에만 쓸 수 있다.
# - 리스트의 원형을 변경한다.

# reversed(리스트)
# - 리스트를 뒤집는다. 
# - 값을 반환한다.
# - 튜플, 스트링, 딕셔너리에도 사용 가능하다.
# - 객체의 원형을 변경하지 않는다.

# 다른 풀이 1
return num_list[::-1]
# 슬라이싱 - [start:end:step] 
# [::-1]: 역순으로 뒤집기

# 다른 풀이 2
result =[]
while(num_list):
    result.append(num_list.pop())
return result
# append(): 새로운 요소를 리스트의 끝에 추가
# pop(): 리스트의 마지막 요소를 제거하고 반환
